flujo/application/core/policies/granular_policy.py

```
@@ -485,6 +520,53 @@ def _is_json_serializable(self, obj: object) -> bool:
+                    # We look for idempotency_key in kwargs or payload
+                    has_key = "idempotency_key" in kwargs
+                    if not has_key and args:
+                        # Check last arg if it's a dict (payload)
+                        last_arg = args[-1]
+                        if isinstance(last_arg, dict) and "idempotency_key" in last_arg:
+                            has_key = True
+                            if last_arg["idempotency_key"] != key:
+                                raise ConfigurationError(
+                                    f"Idempotency key mismatch in tool '{name}': "
+                                    f"expected {key}, got {last_arg['idempotency_key']}"
+                                )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)